systemtest/benchtest/main.go

```diff
@@ -240,3 +236,33 @@ func Run(allBenchmarks ...BenchmarkFunc) error {
 	}
 	return nil
 }
+
+func warmup(agents int, events uint) error {
+	// Assume a base ingest rate of at least 5000 per second, and dynamically
+	// set the context timeout based on this ingest rate, or if lower, default
+	// to 15 seconds. The default 5000 / 5000 = 1, so the default 15 seconds
+	// will be used instead.
+	timeout := time.Duration(math.Max(float64(events/5000), 15)) * time.Second
```

Comment:
We should also consider `max-rate` here. A lower max-rate and corresponding config for `warump-events` can lead to cases were warmup always times out since the base assumption will not align. One option would be to consider `min(5000, maxEPM*60)` instead of hardcoded 5000. WDYT?